Testers/Passes/includes_test/call_structs.txt

```diff
@@ -4,7 +4,7 @@ pass=MustCall function=main var=m.y methods={free}
 pass=MustCall function=main var=km.x methods={free}
 pass=MustCall function=main var=y methods={free}
 pass=CalledMethods function=main var=random_var methods={free}
-pass=CalledMethods function=main var=k.x methods={free}
-pass=CalledMethods function=main var=k.y methods={free}
-pass=CalledMethods function=main var=m.y methods={free}
-pass=CalledMethods function=main var=km.x methods={free}
\ No newline at end of file
+pass=CalledMethods function=main var=k.x methods={creates_obligation,free}
+pass=CalledMethods function=main var=k.y methods={creates_obligation,does_free,free}
```

Comment:
I don't understand why either `k.x` or `k.y` should have `creates_obligation` in their CM set here. That method is only called on `k`, not on its fields.

Reply:
I have it like this because of an implementation detail I should've mentioned sooner; since the IR destructures fields into its elements, internally the called methods/must call set of `k` (referring to the struct in this test case) does not hold any useful information about the resources we need to track. `k` is just a wrapper referring to its fields which actually contain the resources, hence why we treat something like `creates_obligation(k)` as `creates_obligation(k.x, k.y)`

Reply:
This worries me. What if the call was to `free(k)`? If `k.x` or `k.y` had been separately allocated, this would lead to a false negative - so this is a serious problem. For example, consider the following example (from https://stackoverflow.com/questions/13590812/c-freeing-structs):
```
typedef struct Person
{
char * firstname , *last surName;
}Person;
Person *ptrobj =malloc(sizeof(Person)); // memory allocation for struct
ptrobj->firstname = malloc(n); // memory allocation for firstname
ptrobj->surName = malloc(m); // memory allocation for surName

.
. // do whatever you want

free(ptrobj->surName);
free(ptrobj->firstname);
free(ptrobj);
```

If either of the calls to `free(ptrobj->surName)` or to `free(ptrobj->firstname)`, my understanding from what you're saying is that we'd still put `free` in the called methods sets of the fields.